agent/fsx/factory/factory.go

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//    http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package factory
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/credentials"
+	"github.com/aws/amazon-ecs-agent/agent/httpclient"
+	"github.com/aws/aws-sdk-go/aws"
```

Comment:
Please use `goimports` and update the import ordering
```
system packages

amazon-ecs-agent specific packages

external packages
```

Reply:
will do

Reply:
note: there's an agent make target for that: `make goimports`

Reply:
I've addressed all goimports issues